### PR TITLE
1110: Implement LocationIndicatorActive for Memory resource (#571)

### PR DESCRIPTION
**Pull request by jeaaustx**
This is a rebase of https://github.com/ibm-openbmc/bmcweb/pull/571. This commit restructures the original functions for this option.

Note: The Memory "Enabled" option has not yet been rebased to 1110. This commit does not include the changes from the 1060 commit related to that option. When the commit [redfish-core: Memory: Enabled the isolation (aka guard) feature](https://github.com/ibm-openbmc/bmcweb/commit/70c213769052d24244ce58e9836f6f45cedf474c) is rebased the calls to its functions will need to be added to the new locations.

Implement LocationIndicatorActive for Memory resource (#571)

- Add the getLocationIndicatorActive method and the setLocationIndicatorActive method to bmcweb to support the LocationIndicatorActive property for memory.

Tested: Validator passes
```
curl -k -H "X-Auth-Token: $token" -X GET
https://${bmc}/redfish/v1/Systems/system/Memory/dimm0
{
  "@odata.id": "/redfish/v1/Systems/system/Memory/dimm0",
  "@odata.type": "#Memory.v1_11_0.Memory",
  "Id": "dimm0",
  "LocationIndicatorActive": false,
...
}

curl -k -H "X-Auth-Token: $token" -X PATCH -d
'{"LocationIndicatorActive":true}'
https://${bmc}/redfish/v1/Systems/system/Memory/dimm0

curl -k -H "X-Auth-Token: $token" -X GET
https://${bmc}/redfish/v1/Systems/system/Memory/dimm0
{
  "@odata.id": "/redfish/v1/Systems/system/Memory/dimm0",
  "@odata.type": "#Memory.v1_11_0.Memory",
  "Id": "dimm0",
  "LocationIndicatorActive": true,
...
}
```

### CODE REVIEW

File: redfish-core/lib/memory.hpp [hunk 1]
```diff
@@ -33,17 +33,12 @@
 #include <sdbusplus/unpack_properties.hpp>
 
 #include <array>
+#include <functional>
```
Comment (gtmills) [inline redfish-core/lib/memory.hpp:36]: remove this on purpose?

File: redfish-core/lib/memory.hpp [hunk 2]
```diff
@@ -714,152 +709,107 @@ inline void getDimmPartitionData(std::shared_ptr<bmcweb::AsyncResp> asyncResp,
     );
 }
 
-template <typename Handler>
 inline void
-    handleDimmObject(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
-                     const std::string& dimmId, Handler&& handler,
-                     const boost::system::error_code& ec,
-                     const dbus::utility::MapperGetSubTreeResponse& subtree)
+    handleGetDimmData(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
```
Comment (gtmills) [inline redfish-core/lib/memory.hpp:713]: include memory, string

File: redfish-core/lib/memory.hpp (hunk 1 above)
Comment (baemyung) [reply redfish-core/lib/memory.hpp:36]: Additionally, include

```
#include <boost/url/format.hpp>

#include <cstdint>
#include <memory>
#include <string>
#include <vector>
```

Comment (baemyung) [inline redfish-core/lib/memory.hpp:36]: Additionally, several more includes

```
#include <boost/urls/format>

#include <cstdint>
#include <memory>
#include <string>
#include <vector>
```

**Review by baemyung — APPROVED** [review]
It looks good (with a comment to include more headers)

File: redfish-core/lib/memory.hpp [hunk 3]
```diff
@@ -971,55 +993,12 @@ inline void requestRoutesMemory(App& app)
...
-        asyncResp->res.jsonValue["@odata.id"] =
-            boost::urls::format("/redfish/v1/Systems/system/Memory/{}", dimmId);
-
-        getDimmObject(asyncResp, dimmId, getDimmData);
-    });
-
-    BMCWEB_ROUTE(app, "/redfish/v1/Systems/system/Memory/<str>/")
-        .privileges({{"Login"}})
+    BMCWEB_ROUTE(app, "/redfish/v1/Systems/<str>/Memory/<str>/")
+        .privileges(redfish::privileges::patchMemory)
```
Comment (gtmills) [inline redfish-core/lib/memory.hpp:999]: getMemory here

**Review by gtmills — APPROVED** [review]

File: redfish-core/lib/memory.hpp (hunk 1 above)
Comment (jeaaustx) [reply redfish-core/lib/memory.hpp:36]: The boost one is already there on line 30 (although it is using .hpp extension, should be the same thing correct?)

I'll add the other ones.

File: redfish-core/lib/memory.hpp (hunk 2 above)
Comment (jeaaustx) [reply redfish-core/lib/memory.hpp:713]: Added.

**Review by baemyung — APPROVED** [review]
It looks good to me

**Review by gtmills — APPROVED** [review]

**Pull request merged**